mlflow/pyfunc/__init__.py

```diff
@@ -716,6 +719,64 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def _download_model_conda_env(model_uri):
+    model_meta_path = _download_artifact_from_uri(os.path.join(model_uri, MLMODEL_FILE_NAME))
+    model_meta = Model.load(model_meta_path)
+    conda_yml_file_name = model_meta.flavors[FLAVOR_NAME][ENV]
+    return _download_artifact_from_uri(os.path.join(model_uri, conda_yml_file_name))
+
+
+def get_model_dependencies(model_uri, format="pip"):
+    """
+    Given a model URL, and format, return the downloaded dependency file path.
+    Available dependency file format includes "pip" and "conda".
+    """
+    req_file_uri = os.path.join(model_uri, _REQUIREMENTS_FILE_NAME)
+
+    if format == "pip":
+        try:
+            pip_file_path = _download_artifact_from_uri(req_file_uri)
+        except Exception as e:
+            # fallback to download conda.yaml file and parse the "pip" section from it.
+            conda_yml_path = _download_model_conda_env(model_uri)
+            pip_deps = None
+            try:
+                with open(conda_yml_path, 'r') as yf:
+                    conda_yml = yaml.safe_load(yf)
+
+                for dep in conda_yml['dependencies']:
+                    if isinstance(dep, dict) and 'pip' in dep:
+                        pip_deps = dep['pip']
+                        break
+            except Exception as e:
+                raise RuntimeError(
+                    f"Parse conda.yaml file in the model directory failed, error: {repr(e)}."
+                )
+
+            if pip_deps is not None:
+                tmp_dir = tempfile.mkdtemp()
+                pip_file_path = os.path.join(tmp_dir, _REQUIREMENTS_FILE_NAME)
+                with open(pip_file_path, 'w') as f:
+                    f.write("\n".join(dep['pip']) + "\n")
+            else:
+                raise ValueError(
+                    "No pip section found in conda.yaml file in the model directory."
+                )
+        if is_in_databricks_runtime():
+            print(f"On databricks notebook, you can run command '%pip install -r {pip_file_path}' "
+                  "to update current python environment with model's dependencies.")
+        return pip_file_path
+
+    elif format == "conda":
+        conda_yml_path = _download_model_conda_env(model_uri)
+        if is_in_databricks_runtime():
+            print(f"On databricks notebook, you can run command '%conda env update -f {conda_yml_path}' "
+                  "to update current python environment with model's dependencies.")
```

Comment:
`%conda` has been removed in recent versions of the Databricks Machine Learning Runtime. I don't think we should print this.